apbs/create-secret-and-configmap-during-provision.md

```diff
@@ -0,0 +1,44 @@
+# Abstract
+
+This document outlines a proposal to create an additional configmap for services accessible via a mobile client, this configmap would compliment the secret we currently create during the provision of each service via an APB 
+
+## Terms
+
+- Mobile Client (a mobile device running a developers application)
+- APB (ansible playbook bundle)
+
+## Problem Description
+
+Our current approach is to create a secret during the provision of a new mobile service. This secret contains all of the information required to allow the UI to show the service and how to access the service, it also includes any information needed to create the mobile client configuration. This means we expose too much information and also need to filter out any sensitive information (usernames password etc) when constructing our mobile configuration to avoid it being exposed to the client and ending up in the public domain. As each service is different, it often requires a custom piece of code to filter out the fields that we don't want or need exposed. 
+
+## Proposed Solution
+
+When provisioning a new service instance, we should create two configuration resources:
+
+1) A secret containing private information such as usernames and passwords or anything that we do not want to be exposed to the mobile client.
+2) A configmap containing all of the required information needed for a mobile client to use the service.
+
+Note it may be neccessary to duplicate some information across each of these resources (ie the URI of the service for example).
+
+Each of these resources should have at least the following information and labels:
+
+- Labels: 
+    - **mobile:enabled**. Will give a mechanism to filter these resources from other resources of the same type from the UI and CLI
+    - **serviceName:< serviceName >**. Depending on whether we want to allow more than one service of a particular kind to be provisioned to a namespace, this will allow us to identify which service the information is for without needing to name the resource itself after the service.
+
+- Data
+    - **URI** both the configmap and secret will require this. 
```

Comment:
for services that will have other config options, I think it is better to group them together and give them a key called `config`, rather than having them at the root of the configmap object.

e.g.

```
{
  mobile: 'enabled',
  serviceName: 'push',
  URI: 'https://...',
  config: {
    key:"...",
    secret: "...."
  }
}
```

Reply:
The key values should all be configuration in the configmap, and the labels then (which wont show up in the config objects) give the meta data.

Reply:
also probably each each config should have the same format, e.g. each extra config should have `name` and `value` fields. This is to make it easier to lookup and manage the config options.